src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -915,14 +917,28 @@ def _add_problem_span(self, span: Span) -> None:
         self.consecutive_db_spans.append(span)
 
     def _validate_and_store_performance_problem(self):
+        independent_db_spans = self._find_independent_spans(self.consecutive_db_spans)
         exceeds_count_threshold = len(self.consecutive_db_spans) >= self.settings.get(
             "consecutive_count_threshold"
         )
-        independent_db_spans = self._find_independent_spans(self.consecutive_db_spans)
+        exceeds_span_duration_threshold = all(
+            get_span_duration(span).total_seconds() * 1000
+            > self.settings.get("span_duration_threshold")
+            for span in independent_db_spans
+        )
+        exceeds_total_duration_threshold = exceeds_duration_threshold = self._sum_span_duration(
+            self.consecutive_db_spans
+        ) > self.settings.get("total_duration_threshold")
         exceeds_duration_threshold = self._sum_span_duration(
             independent_db_spans
         ) > self.settings.get("duration_threshold")
```

Comment:
🤔 what's the significance of checking both the independent span durations and the consecutive span durations?

Reply:
Good question, (and if my logic is off here lmk)
In a set of consecutive spans, the independent spans are the ones that we think can be in parallel. If an independent span take almost as much as the consecutive spans, then there isn't really benefit in parallelizing the query. 
Now there is going to be cases with multiple independent spans in a set of consecutive db spans, for this case this logic doesn't really work, so in the future a cost function will be introduced to help evaluate if its "worth" running the spans in parallel.

Reply:
That makes sense! Would you mind adding a comment to explain this? It also might help to give the thresholds more descriptive names that explain their purpose 🙏🏻

Reply:
Yeah, for sure!